snapshot/git/gitdb/backends.go

```diff
@@ -0,0 +1,62 @@
+package gitdb
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/scootdev/scoot/snapshot"
+)
+
+type backend interface {
+	parseID(id string, kind string, parts []string) (snap, error)
+}
+
+type uploader interface {
```

Comment:
Yeah, it's weird, but the alternatives are weirder.

Option 1: Download is on backend, not snap.

Look at checkout.go, where it downloads before checkout: it would be weird to have to get the right backend to pass right after you parsed it.

Option 2: Upload on Value.
Well, it then has to figure out what the right place to upload is, etc.

So, because upload and download are the reverses of one another, they start from opposite places.